vllm/model_executor/layers/attention.py

```diff
@@ -334,6 +334,19 @@ def __init__(
                 self.rotary_emb = DynamicNTKScalingRotaryEmbedding(
                     head_size, rotary_dim, max_position, base, is_neox_style,
                     scaling_factor)
+            elif scaling_type == "yarn":
+                new_max_position = rope_scaling[
```

Comment:
Is the variable name `new_max_position`? If I understand correctly, this is the original maximum length (e.g., 4096 for `NousResearch/Yarn-Llama-2-7b-64k`).

Reply:
Let me take a look at both comments!